### PR TITLE
htsget-client: Add a simple htsget client

### PR DESCRIPTION
**Pull request by kemp-google**
This is a simple and intentionally somewhat verbose htsget client that uses
OAuth2 application default credentials to authenticate itself.

### CODE REVIEW

File: htsget-client/main.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+		return fmt.Sprintf("%d KB", kb)
+	}
+	return fmt.Sprintf("%d bytes", n)
+}
+
+func fetchBlob(ctx context.Context, target string, headers map[string]string) (io.ReadCloser, error) {
+	if v := strings.TrimPrefix(target, "data:"); v != target {
+		parts := strings.Split(v, ",")
+		if len(parts) != 2 {
+			return nil, errors.New("malformed data URl")
```
Comment (gkelly) [inline htsget-client/main.go]: URL

Comment (kemp-google) [reply htsget-client/main.go]: Done

File: htsget-client/main.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+				TLSClientConfig: &tls.Config{
+					RootCAs: pool,
+				}},
+		})
+		log.Printf("Using CA override bundle from %q", bundle)
+	}
+
+	client, err := google.DefaultClient(ctx, scope)
+	if err != nil {
+		log.Fatalf("Failed to creating client: %v", err)
```
Comment (gkelly) [inline htsget-client/main.go]: "Failed to create"

Comment (kemp-google) [reply htsget-client/main.go]: Done

**Pull request merged**